graylog2-server/src/main/java/org/graylog2/shared/rest/CORSFilter.java

```diff
@@ -29,7 +29,11 @@
     public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext) throws IOException {
         // we have already added the necessary headers for OPTIONS requests below
         if (requestContext.getRequest().getMethod().equalsIgnoreCase("options")) {
-            return;
+            if(Response.Status.Family.familyOf(responseContext.getStatus()) == Response.Status.Family.INFORMATIONAL) {
+                return;
+            }
+            responseContext.setStatus(Response.Status.NO_CONTENT.getStatusCode());
+            responseContext.setEntity("");
         }
```

Comment:
The `return` is there to prevent double response headers for 20x responses. From what I've seen, Jersey is not adding the headers for non-20x responses. Which responses did you encounter that had double headers?

UPDATE: I just noticed that it's the wrong Response family I am using to compare. Fixing.